File: ScoreSystemServer/ScoreSystemServer/src/main/java/com/score/system/entity/request/RegisterRequest.java
package com.score.system.entity.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class RegisterRequest {
    @NotBlank(message = "用户名不能为空")
    @Size(min = 3, max = 20, message = "用户名长度必须在3到20之间")
    @Pattern(regexp = "^[a-zA-Z0-9_]+$", message = "用户名只能包含字母、数字和下划线")
    private String username;

    @NotBlank(message = "密码不能为空")
    @Size(min = 6, max = 32, message = "密码长度必须在6到32之间")
    private String password;

    @NotBlank(message = "姓名不能为空")
    @Pattern(regexp = "^[\\u4e00-\\u9fa5a-zA-Z\\s]{2,30}$", message = "姓名只能包含中文、字母、空格，长度2-30位")
    private String name;
    private Integer level;
    private Integer role;
}

File: ScoreSystem/ScoreSystem/ScoreSystem/Model/RegisterRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScoreSystem.Model
{
    public class RegisterRequest
    {
        private string username;
        private string password;
        private string name;
        private int level;
        private int role;

        public string Username { get => username; set => username = value; }
        public string Password { get => password; set => password = value; }
        public string Name { get => name; set => name = value; }
        public int Level { get => level; set => level = value; }
        public int Role { get => role; set => role = value; }
    }
}
